lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -533,8 +530,8 @@ public static function get_user_data() {
 				 * @param WP_Theme_JSON_Data_Gutenberg Class to access and update the underlying data.
 				 */
 				$theme_json = apply_filters( 'wp_theme_json_data_user', new WP_Theme_JSON_Data_Gutenberg( $config, 'custom' ) );
-				$config     = $theme_json->get_data();
-				return new WP_Theme_JSON_Gutenberg( $config, 'custom' );
+
+				return $theme_json->get_theme_json();
```

Comment:
A bit below, there's another `WP_Theme_JSON_Data_Gutenberg` call but we cannot update it as we do on this one. That code is different than the core code and we need to consolidate both. That's a follow-up PR that shouldn't prevent this one from landing.

Reply:
Perhaps we can absorb `$config['isGlobalStylesUserThemeJSON'] = true;` as part of `WP_Theme_JSON_Data_Gutenberg`. We may want to still check that's true before returning to be extra-safe (line 553) but we wouldn't need a second transformation.

Reply:
@oandregal I observed it but presumed that GB and Core might be doing things differently here.  I'll look at how to incorporate that in a different PR.

Reply:
Looks like the `$config['isGlobalStylesUserThemeJSON']` modification was made as part of 2012a36 (#58409). 

@ajlende is there a reason why this would still be necessary if we use the `WP_Theme_JSON_Gutenberg` object already present in the `WP_Theme_JSON_Data_Gutenberg` object?

Reply:
The `$config['isGlobalStylesUserThemeJSON']` is required for migrations that don't need to run for the custom origin in `WP_Theme_JSON_Schema::migrate` (called on `new WP_Theme_JSON()`). If the origin could be passed to `migrate` in some other way, then it would be fine.

https://github.com/WordPress/gutenberg/blob/cc2fcbe5e7d4384c27aa6615d2a9660f873fe572/lib/class-wp-theme-json-schema-gutenberg.php#L115-L124

Reply:
Added a draft PR to explain what I'm talking about https://github.com/WordPress/gutenberg/pull/62305

Reply:
Thanks @ajlende! I'll try to review this later this week.

Reply:
Approved #62305 which improves the API and should bring back this change. Thanks for the follow-up, Alex!